lib/ui/platform_dispatcher.dart

```diff
@@ -2605,3 +2621,37 @@ class SemanticsActionEvent {
     );
   }
 }
+
+/// Callback to subscribe to [ViewFocusChange] notifications.
+typedef ViewFocusChangeCallback = void Function(ViewFocusChange);
+
+/// An event to communicate view focus changes between the framework and engine.
```

Comment:
nit: since this object is used both to request focused as well as to notify about a focus change, it is no longer an "event". Events express things that happen that a component of the system needs to react to, or deal with. They are not used to request something.

Possible alternatives:

1. Do not use this type for the framework to request a focus change. Instead, unwrap the constructor parameters and pass them directly, e.g. `PlatformDispatcher.requestViewFocus(int viewId, ViewFocusState state)`.
2. Rename this object to something that does not imply "event" or "change". I'm struggling with thinking of something, possibly because there's no real reason to have a type for the focus change event, but maybe I lack imagination.

Reply:
> nit: since this object is used both to request focused as well as to notify about a focus change, it is no longer an "event". 

Well it depends on the mental model behind it. My initial idea was that the framework and engine would communicate  focus changes and assume the other part would react accordingly hence the "notification system". What you're proposing is not that different both imples that the engine know what the framework should do and viceversa. I feel like having these messages passed without explicitly communicating intent plays better with the multi platform nature of flutter (you platform, there is this focus change event... figure out the right treatment). 

Open to change the model to a more imperative/explicit intent request tho. WDYT?

Reply:
The reason why I think notification is not the appropriate paradigm in the reverse case is because a notification has the semantics along the lines of "event A happened, now you react to it". There's the _fact_ of "event A happened", then there's a notification about the fact.

This semantic holds for the case when the engine notifies the framework. The browser passes focus to a Flutter view, and the framework now may react to it, if it wants to. The browser is not requesting focus. It already focused on the view. The engine is simply stating the fact in its notification. If framework does nothing, the view will still be focused.

However, the opposite direction is different. The framework did not change the view focus. Instead, it is requesting that the engine change the focus of the view. If the engine does nothing, the view focus will not change.

So I still think a method makes more sense when the framework requests that the engine change focus. But a notification/event/message style API makes sense in the engine-to-framework direction.

Reply:
> However, the opposite direction is different.  The framework did not change the view focus.

To me, they are simmetric. 

If it needs to notify the engine, it actually did. It is indeed a notification to let the engine know that the focus moved from widget `A` (focus node to be more precise) in a given flutter view to widget `B` that's located in a different flutter view. If the framework notion of focus changes across flutter views, it needs to notify the engine so that it can sync the platform (browser) one with it (by moving the browser focus to that given flutter view).

Frame 1:
```
- FlutterView1 (has focus)
  - Widget 1  [focused]
- FlutterView2 (no focus)
  - Widget 1 [not focused]
```

Something happens, Frame 2: 
```
- FlutterView1 (no focus)
  - Widget 1  [not focused]
- FlutterView2 (has focus)
  - Widget 1 [focused]
```

Reply:
>> The framework did not change the view focus.
> If it needs to notify the engine, it actually did

There are two notions of focus playing in the same space, they need to be reconciled when one changes.
Or to be more precise a given flutter view has 2 notions of focus that need to be in sync: the flutter and platform one.

Reply:
My mental model of this was also that the two directions are slightly different: The engine informs the framework that view focus has changed while the framework asks (or demands) the engine to change the view focus. The framework doesn't have the power to change view focus, it can only ask for it or respond to changes in it. I think it is worthwhile to also respect this semantic difference in the API design.

(See also my longer comment above.)

Reply:
Thank you for the detailed feedback. I now agree with you both and made the requested changes. PTAL